### PR TITLE
Introduce PrimitiveValueObject for single-property ValueObjects

### PR DESCRIPTION
**Pull request by MikelThief**
This PR is realization of the need represented under issue #175.

### CODE REVIEW

**Comment by altmann** [comment]
I also have such an abstract value object in my codebase in my current project. It would be great of you also implement ToString () and IComparable.

File: CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs [hunk 1]
```diff
@@ -28,7 +28,17 @@ public void Two_VO_of_the_same_content_are_equal()
         }
 
         [Fact]
-        public void It_is_possible_to_override_defaul_equality_comparison_behavior()
+        public void Two_PVO_of_the_same_content_are_equal()
+        {
+            var emailAddress1 = new EmailAddress("a@b.com");
+            var emailAddress2 = new EmailAddress("a@b.com");
+
+            emailAddress1.Equals(emailAddress2).Should().BeTrue();
```
[the remark this answers is not in the document]
Comment (MikelThief) [reply CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs]: Exactly, I valued consistency over correctness. @vkhorikov what do you think?

File: CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs [hunk 2]
```diff
@@ -46,6 +56,15 @@ public void Comparing_value_objects_of_different_types_returns_false()
             vo1.Equals(vo2).Should().BeFalse();
         }
 
+        [Fact]
+        public void Comparing_primitive_value_objects_of_different_types_returns_false()
```
[the remark this answers is not in the document]
Comment (MikelThief) [reply CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs]: This is again consistency-related issue. See existing `Comparing_value_objects_of_different_types_returns_false`

**Comment by MikelThief** [comment]
@altmann `ToString()` implementation will basically route that call to `Value.ToString()` If `Value` does not have meaningful conversion to string, then such a method in `PrimitiveValueObject` gives you fake conviction about retrieving a string from value object. You can always call `someVO.Value.ToString()`.

About `IComparable`, how would you see it implemented for an arbitrary T?

**Comment by altmann** [comment]
@MikelThief 

Exactly, ToString() implementation should route to Value.ToString(). I understood that the class PrimitiveValueObject is used only for PRIMITIVE types (string, int, double, DateTime, ...) and not for complex types. So routing to Value.ToString() should be works fine. If you need some special Logic then you can override the method. 

SomeVO.Value.ToString() can be called but I don't like it because it show some implementation details. You should use the Value property as rare as possible. In a perfect world the Value property should be private but this ist not realistic. 

I think all primitive types implemented IComparable. So making an generic constraint to this interface would be an valid option.

File: CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vkhorikov) [reply CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs]: Please use `emailAddress1.Should().Be(emailAddress2)`. Even better if you could fix the other occurrences too.

File: CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vkhorikov) [reply CSharpFunctionalExtensions.Tests/ValueObjectTests/BasicTests.cs]: Please fix everywhere to `values` for consistency.

**Comment by MikelThief** [comment]
@altmann 

The issue and related PR were touching a subject of writing noticeable amount of boilerplate code while creating a single-property ValueObject. I was not entirely confident about the name, hence `SinglePropertyValueObject` proposition was also made (which is clearly states the intention in awful way). 
`DateTime` is not primitive data type. I think this would turn into `IComparableValueObject` which by its name goes next to `SinglePropertyValueObject` (but it is a little better than my idea for a name).
By proposing `Primitive...` I was thinking about simplicity of a concept (where `Value` most likely keeps a primitive type but it is not limited to. It could also be `Simple...` but I not fond of `Simple...` names. I often need to peek what's inside... and this is not simple.

About `ToString()` - I would really like somebody else to express their opinion as well. I am now inclined into implementing it. After all it will work the same as for all other types.

**Comment by altmann** [comment]
@MikelThief Today I played a little bit with IComparable and ToString(). 

You are right with IComparable. The problem which I faced is that Nullable<T> doesn't implement IComparable. 

The ToString() implemenation routing to Value.ToString() works great. These are only some ideas. If these ideas don't fit in your concepts then ignore it.

File: CSharpFunctionalExtensions/ValueObject/PrimitiveValueObject.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+
+namespace CSharpFunctionalExtensions
+{
+    public abstract class PrimitiveValueObject<T> : ValueObject
```
[the remark this answers is not in the document]
Comment (vkhorikov) [reply CSharpFunctionalExtensions/ValueObject/PrimitiveValueObject.cs]: I like the `SimpleValueObject` one. I'm merging this commit and renaming the class to SimpleValueObject. @MikelThief let me know if you have any objections

[the remark this answers is not in the document]
Comment (MikelThief) [reply CSharpFunctionalExtensions/ValueObject/PrimitiveValueObject.cs]: I like `Singular...` but can cope with `Simple...` as well. Thanks!

**Pull request merged**